### DISCUSSION TITLE
Feasibility of Running CUBIN on AMD GPUs

### DISCUSSION THREAD

**Discussion by ykdu** [vosen/ZLUDA#291] [General]
I know that SASS isn't fully documented, but there’s a list of disassembled instructions available in [NVIDIA's CUDA documentation](https://docs.nvidia.com/cuda/cuda-binary-utilities/index.html#volta-instruction-set). However, I’m not sure if that’s a complete list of instructions or just a partial reference.

Does anyone have insights on whether building a compiler or translator that maps NVIDIA GPU ISA to AMD GPU ISA or AMD LLVM IR is technically feasible? What kind of challenges should we expect, especially given the architectural differences between NVIDIA and AMD GPUs? Is there any project out there that’s already working on something similar?

I’d love to hear thoughts or experiences!

**Comment by vosen** [comment]
It's not impossible, that's what Nintenfo Switch emulators do
